src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/Utilities.cs

```diff
@@ -88,37 +88,94 @@ internal class ScriptingRuntimeHelpers
     {
         public static object Int32ToObject(int i)
         {
+            switch (i)
+            {
+                case -1:
+                    return Int32_m;
+                case 0:
+                    return Int32_0;
+                case 1:
+                    return Int32_1;
+                case 2:
+                    return Int32_2;
+                case 3:
+                    return Int32_3;
+            }
+
             return i;
         }
+        
+        internal static readonly object Int32_m = -1;
+        internal static readonly object Int32_0 = 0;
+        internal static readonly object Int32_1 = 1;
+        internal static readonly object Int32_2 = 2;
+        internal static readonly object Int32_3 = 3;
```

Comment:
I assume we're doing this to avoid the repeated boxing costs?  If so, how did we arrive at doing this for the range [-1,3] and not a larger range?